### ISSUE TITLE
VSCode: port or alternative

### ISSUE THREAD

**Issue by asiloisad** [nteract/hydrogen#2155]
Hi,
in last days the Atom software look like nearly dead. The hydrogen package is one of the most important packages for my everyday work and now I'm looking for something similar in VSCode.

Have you seen any similar package to Hydrogen in VSCode or are there any plans to port the package? The `.ipynb` workflow is good, but not nearly good as idea od Hydrogen.

Best regards

**Comment by pintergreg** [comment]
It's now officially dead: https://github.blog/2022-06-08-sunsetting-atom/

**Comment by asiloisad** [comment]
at first the bubbles results was impossible to implement in vscode, but 2-3 years ago the posibillity was added. just check https://www.julia-vscode.org/ and see it is possible.

**Comment by matthewgdv** [comment]
Agreed, a port to either VSCode or Pycharm would be amazing. My department relies on Hydrogen's interactive workflows pretty extensively and having to go back to traditional jupyter notebooks will seriously suck :(

**Comment by okenakt** [comment]
Atom+Hydrogen is still a powerfull and no alternative development tool.
The flexibility to change execution blocks by selection without cell definitions like jupyter, and coding as a pure python script with inline debugging is great interactive experience.
No longer possible to handle high-dimensional arrays or huge datatables without Hydrogen.

**Comment by juliuscrn** [comment]
> VsCode's Jupyter integration comes close if that works for you. https://code.visualstudio.com/docs/datascience/jupyter-notebooks

Sorry to be nitpicky, but it really doesn't. What Microsoft did with that addon is one big load of nonsense. The very concept of adding some weird characters to turn it into a "cell" that can then be run is just stupid beyond belief.

**Comment by ttxtea** [comment]
I am not sure how nteract notebooks work, but I think they are normal json format and not as in hydrogen, normal python code plus comments.

The json way as severe downsides
- hard to read and fix in a normal fall back editor like vim
- therefore not suitable for normal git repos, where you would like to merege and fix files by hand
- you cant just run it with python as a script 

all of these issues are perfectly handeled by atom+hydrogen. Visualstudio notebooks dont do any of that.

I would be wonderful to find a way to continue hydrogen in some way. Either as part of a striped and simplified atom fork that is sustainable by a community. Or by some new editor that is simply a hydrogen frontend. But I don't have much hope. Its back to VIM for me.

**Comment by asiloisad** [comment]
Has anyone heard of a good alternative? has there been anything noteworthy recently? as many people mentioned above, jupyter and .ipynb is not a solution. I'm also aware of the developing Pulsar, but it has a lot of problems with hydrogen support.

**Comment by ttxtea** [comment]
[Pulsar](https://pulsar-edit.dev/) is a very active `atom` community fork. Packages like `hydrogen` can be installed. I did have some problems, but I heard of people where it seems to work just like in good old `atom`.

**Comment by chriskuchar** [comment]
This is the very reason why I love open source, rather than software as a service. Atom gets sunsetted, and people who care about atom and atom + hydrogen just fork a new one. You guys at Pulsar are amazing.

**Comment by contang0** [comment]
Agreed. Isn't it amazing that with all the development effort Microsoft spends on their VS Code Python extension it's still so vastly inferior to Atom+Hydrogen? I filled out their surveys countless of times pointing to its deficiencies, nothing ever improved, and probably never will.

**Comment by okenakt** [comment]
Hi,

In the past few years, AI coding has been transforming development itself, but I still need the features and experience of Hydrogen.

So I (mostly Claude) created [a similar extension](https://marketplace.visualstudio.com/items?itemName=okenakt.protium) for VS Code.
Please give it a try!

**Comment by asiloisad** [comment]
There is fork of hydrogen called [hydrogen-next](https://github.com/asiloisad/pulsar-hydrogen-next) which works fine in Atom, Pulsar and PulsarNext.

**Comment by humanplayer2** [comment]
For Python, [vscode-jupyter-python](https://marketplace.visualstudio.com/items?itemName=kylebarron.vscode-jupyter-python) works excellently for me.
